spec/features/admin_claim_approve_spec.rb

```diff
@@ -38,6 +38,18 @@
         )
       end
     end
+
+    context "When the payroll gender is missing" do
```

Comment:
Yeah I thought about that too, my thinking was that this card is all about informing a service operator that a claim cannot be approved more than preventing them. I think we can trust service operators to not edit the HTML of a page?

Perhaps I should reword the spec?

Reply:
I know what you mean, I guess I'm just always a bit iffy about relying on the frontend to control validations, even if there is only a tiny risk that something will make it through

Reply:
So it would be belt-and-braces to do the defensive coding to 100% prevent a claim missing the required value for payroll_gender, but it's good to measure the cost of doing this extra work against a) the risk that the thing we are protecting against happening, and b) the cost if it did happen. In this case I think both the risk, and the cost is low – it's unlikely a service operator is going to be hacking URLs, and if they did, we'd generate a payroll file with an invalid gender, which would either be spotted by us when we eyeball the file (whilst we're doing that), or by Cantium when they receive it. In both cases, we can recover from this situation without too much reputational or operational cost.

Reply:
@robbiepaul How much of a pain is it to add a validation at this point?

Reply:
Do you mean model validation? My thinking was:

Adding another validation context (eg `:approval`) for just one field could be confusing, so we'd probably want to add `:approval` to all the fields required at that stage in the same way we do for `:submit`? The only thing we really want to check at this point if `payroll_gender` is missing so that seemed excessive.

Also the validation would only need to happen on approval and not rejection so we'd need to only validate when `result == "approved"`

If you feel strongly about it, I can have a look at implementing it but I feel like if someone somehow managed to get a claim approved with a missing payroll gender we'd be notified by Cantium at which point we could look at what happened and come up with a solution.

Reply:
We could potentially do a one line validation in the controller? Something like:
```
return redirect_to admin_claim_path(@claim), alert: "Claim cannot be approved" if params[:result] == "approved" && @claim.payroll_gender_missing?
```

Reply:
Yeah, I reckon that would be enough 👍 Just keep it super simple

Reply:
I've added that line and included specs 👍